templates/components/agents/python/financial_report/app/workflows/financial_report.py

```
@@ -24,19 +24,27 @@
 )
 
 
+def _create_query_engine_tool(params=None) -> QueryEngineTool:
+    if params is None:
+        params = {}
+    # Add query tool if index exists
+    index_config = IndexConfig(**params)
+    index = get_index(index_config)
+    if index is None:
+        return None
+    return get_query_engine_tool(
```

Ruff (0.8.0)

Undefined name `get_query_engine_tool`

(F821)